components/server/src/api/envvar-service-api.ts

```diff
@@ -0,0 +1,243 @@
+import { EnvironmentVariableService as EnvironmentVariableServiceInterface } from "@gitpod/public-api/lib/gitpod/v1/envvar_connect";
+import {
+    ListUserEnvironmentVariablesRequest,
+    ListUserEnvironmentVariablesResponse,
+    UpdateUserEnvironmentVariableRequest,
+    UpdateUserEnvironmentVariableResponse,
+    DeleteUserEnvironmentVariableRequest,
+    DeleteUserEnvironmentVariableResponse,
+    CreateUserEnvironmentVariableRequest,
+    CreateUserEnvironmentVariableResponse,
+    ListConfigurationEnvironmentVariablesRequest,
+    ListConfigurationEnvironmentVariablesResponse,
+    UpdateConfigurationEnvironmentVariableRequest,
+    UpdateConfigurationEnvironmentVariableResponse,
+    EnvironmentVariableAdmission,
+    CreateConfigurationEnvironmentVariableRequest,
+    CreateConfigurationEnvironmentVariableResponse,
+    DeleteConfigurationEnvironmentVariableRequest,
+    DeleteConfigurationEnvironmentVariableResponse,
+    ResolveWorkspaceEnvironmentVariablesResponse,
+    ResolveWorkspaceEnvironmentVariablesRequest,
+    ResolveWorkspaceEnvironmentVariablesResponse_EnvironmentVariable,
+} from "@gitpod/public-api/lib/gitpod/v1/envvar_pb";
+import { inject, injectable } from "inversify";
+import { EnvVarService } from "../user/env-var-service";
+import { PublicAPIConverter } from "@gitpod/gitpod-protocol/lib/public-api-converter";
+import { ProjectEnvVarWithValue, UserEnvVar, UserEnvVarValue } from "@gitpod/gitpod-protocol";
+import { WorkspaceService } from "../workspace/workspace-service";
+
+@injectable()
+export class EnvironmentVariableServiceAPI implements ServiceImpl<typeof EnvironmentVariableServiceInterface> {
+    @inject(EnvVarService)
+    private readonly envVarService: EnvVarService;
+
+    @inject(WorkspaceService)
+    private readonly workspaceService: WorkspaceService;
+
+    @inject(PublicAPIConverter)
+    private readonly apiConverter: PublicAPIConverter;
+
+    async listUserEnvironmentVariables(
+        req: ListUserEnvironmentVariablesRequest,
+        context: HandlerContext,
+    ): Promise<ListUserEnvironmentVariablesResponse> {
+        const response = new ListUserEnvironmentVariablesResponse();
+        const userEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+        response.environmentVariables = userEnvVars.map((i) => this.apiConverter.toUserEnvironmentVariable(i));
+
+        return response;
+    }
+
+    async updateUserEnvironmentVariable(
+        req: UpdateUserEnvironmentVariableRequest,
+        context: HandlerContext,
+    ): Promise<UpdateUserEnvironmentVariableResponse> {
+        if (!req.envVarId) {
+            throw new ConnectError("envVarId is not set", Code.InvalidArgument);
+        }
+
+        const response = new UpdateUserEnvironmentVariableResponse();
+
+        const userEnvVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
```

Comment:
Why do we need to fetch everything again?  typorm supports partial updates, we should just partial change  to DB layer and add DB tests for env-var-service to check that it works, see for instance for update org settings

Reply:
i.e. what we discussed on teaming yesterday, use repo.update + repo.findOne(id)